src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -1786,5 +2023,639 @@ static void Main()
+                // (8,17): error CS0030: Cannot convert type 'byte[]' to 'C3'
+                //         var z = (C3)"cat"u8;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, @"(C3)""cat""u8").WithArguments("byte[]", "C3").WithLocation(8, 17)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void UserDefinedExplicitConversions_03()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        C1 x = ""hello""u8;
+        C2 y = ""dog""u8;
+        C3 z = ""cat""u8;
+    }
+}
+
+class C1
+{
+    public static explicit operator C1(byte[] x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static explicit operator C2(Span<byte> x)
+    {
+        return new C2();
+    }
+}
+
+class C3
+{
+    public static explicit operator C3(ReadOnlySpan<byte> x)
+    {
+        return new C3();
+    }
+}";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            comp.VerifyDiagnostics(
+                // (7,16): error CS0266: Cannot implicitly convert type 'byte[]' to 'C1'. An explicit conversion exists (are you missing a cast?)
+                //         C1 x = "hello"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConvCast, @"""hello""u8").WithArguments("byte[]", "C1").WithLocation(7, 16),
+                // (8,16): error CS0029: Cannot implicitly convert type 'byte[]' to 'C2'
+                //         C2 y = "dog"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""dog""u8").WithArguments("byte[]", "C2").WithLocation(8, 16),
+                // (9,16): error CS0029: Cannot implicitly convert type 'byte[]' to 'C3'
+                //         C3 z = "cat"u8;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, @"""cat""u8").WithArguments("byte[]", "C3").WithLocation(9, 16)
+                );
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void UserDefinedExplicitConversions_04()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var x = (C1)""hello""u8;
+    }
+}
+
+class C1
+{
+    public static explicit operator C1(byte[] x)
+    {
+        Helpers.Print(x);
+        return new C1();
+    }
+}
+";
+            var comp = CreateCompilation(source + HelpersSource, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"
+{ 0x68 0x65 0x6C 0x6C 0x6F }
+").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void UserDefinedExplicitConversions_05()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        var y = (C2)""dog""u8;
+        var z = (C3)""cat""u8;
```

Comment:
> I didn't understand why an error is produced here. 

You are correct. This doesn't follow the proposed specification. I guess, I forgot to add a PROTOTYPE comment about that in the Binder. I decided to go with ```byte[]``` as the natural type for now. That allows us to get away without special conversion rules for UTF8 string literals (do not confuse with string constant expressions, UTF8 string literals are not constants) because there are  already implicit user-defined conversions from ```byte[]``` to ```Span<byte>``` and ```ReadOnlySpan<byte>```, and there are no existing conversions from ```ReadOnlySpan<byte>``` to ```byte[]``` or ```Span<byte>```. It feels like the language rules are simpler. Also, even when we are creating a ```ReadOnlySpan<byte>```, we are creating a ```byte[]``` first. It feels like ```byte[]``` is a better choice for a natural type.